src/fmt.jl

```diff
@@ -29,7 +29,10 @@ default_spec!(::Type{T}, ::Type{K}) where {T,K} =
     (DEFAULT_FORMATTERS[T] = DEFAULT_FORMATTERS[K]; nothing)
 
 # seed it with some basic default formatters
-for (t, c) in [(Integer,'d'), (AbstractFloat,'f'), (AbstractChar,'c'), (AbstractString,'s')]
+ComplexInteger = Complex{T} where T<:Integer
+ComplexFloat = Complex{T} where T<:AbstractFloat
+for (t, c) in [(Integer,'d'), (AbstractFloat,'f'), (AbstractChar,'c'), (AbstractString,'s'),
+    (ComplexInteger, 'd'), (ComplexFloat, 'f'), (Number,'S'), (AbstractIrrational,'S')]
```

Comment:
You don't need both `Number` and `AbstractIrrational`, since `AbstractIrrational` is a subtype of `Number`.
You could simply add `(Number, 's')`.  For complex, just add `(Complex, 's')`, and have a special `_srepr` function for *any* complex number.

Reply:
I introduced both because I wanted to allow the user to set different default formats. E.g. in many cases users will want to display a floating point number for `AbstrqactIrrational`s.
This can be done by calling `fmt_default!(AbstractIrrational, 'f')`. Other Number types won't be affected and still display as strings.